packages/wmr/src/plugins/wmr/client.js

```diff
@@ -164,14 +164,34 @@ export function style(filename, id) {
 	}
 }
 
+function traverseSheet(sheet, target) {
+	for (let i = 0; i < sheet.rules.length; i++) {
+		if (sheet.rules[i].href && resolve(strip(sheet.rules[i].href)) === strip(target)) {
+			return sheet.rules[i];
+		} else if (sheet.rules[i].styleSheet) {
+			return traverseSheet(sheet.rules[i].styleSheet, target);
+		}
+	}
+}
+
 // Update a non-imported stylesheet
 function updateStyleSheet(url) {
 	const sheets = document.styleSheets;
+
 	for (let i = 0; i < sheets.length; i++) {
 		if (sheets[i].href && strip(sheets[i].href) === url) {
 			// @ts-ignore
 			sheets[i].ownerNode.href = strip(url) + '?t=' + Date.now();
 			return true;
 		}
+
+		const found = traverseSheet(sheets[i], url);
+		if (found) {
+			const index = [...found.parentStyleSheet.rules].indexOf(found);
+			const keyword = found.cssText.includes('@import') ? '@import' : '@use';
+			found.parentStyleSheet.insertRule(`${keyword} url("${strip(url) + '?t=' + Date.now()}")`, index);
```

Comment:
This should preserve media and `@supports` queries:

```suggestion
			const urlStr = JSON.stringify(strip(url) + '?t=' + Date.now());
			const css = found.cssText.replace(/^(@import|@use)\s*(?:url\([^)]*\)|(['"]).*?\2)/, '$1 ' + urlStr);
			found.parentStyleSheet.insertRule(css, index);
```